File: KeepIt/app/src/main/java/com/utbm/KeepIt/LoginActivity.java
package com.utbm.KeepIt;

import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import DAO.LoginDao;

public class LoginActivity extends AppCompatActivity {

    //定义控件
    private EditText username , pwd ;
    private Button loginbtn;
    private LoginDao dao;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        dao = new LoginDao(this) ;

        //获取控件
        username = (EditText) findViewById(R.id.activity_login_et_username);
        pwd = (EditText) findViewById(R.id.activity_login_et_pwd);
        loginbtn = (Button) findViewById(R.id.activity_login_btn);
        loginbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                //判断登陆成功与否
                if(dao.login(username.getText().toString(),pwd.getText().toString())){
                    //登陆成功页面跳转
                    Intent intent = new Intent(LoginActivity.this, MainActivity.class);
                    intent.putExtra("username",username.getText().toString());
                    startActivity(intent);
                }else {
                    //登陆失败，显示提示信息
                    Toast.makeText(LoginActivity.this, "用户名和密码错误", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: LO52_2019_Equal/Project/app/src/main/java/com/startup42/equal/view/LoginActivity.kt
package com.startup42.equal.view

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import com.startup42.equal.R
import com.startup42.equal.viewModel.LoginViewModel

import kotlinx.android.synthetic.main.activity_login.*


class LoginActivity : AppCompatActivity() {

    val viewModel = LoginViewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        loginButton.setOnClickListener {
            if (!emailInput.text.isEmpty() and !passwordInput.text.isEmpty()) {

                val hashMap = HashMap<String, Any>()
                hashMap.put("email", emailInput.text.toString())
                hashMap.put("password", passwordInput.text.toString())

                loginButton.isEnabled = false
                viewModel.sendData(hashMap)
                    .doOnNext {

                        if (it == "You are logged") {
                            val intent = Intent(this@LoginActivity, HomeActivity::class.java)
                            intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                            startActivity(intent)
                            finish()
                        } else {
                            this.runOnUiThread{
                                loginButton.isEnabled = true
                                Toast.makeText(this, it, Toast.LENGTH_LONG).show()
                            }
                        }
                    }
                    .doOnError { it.printStackTrace() }
                    .doOnComplete{ println("onComplete!") }
                    .subscribe()

            } else {
                //TODO set string
                Toast.makeText(this, "TODO", Toast.LENGTH_LONG).show();
            }
        }

        notRegisteredYetLink.setOnClickListener{
            val intent = Intent(this, RegisterActivity::class.java)
            startActivity(intent)
        }
    }

}
